src/components/AuthenticationModal/AuthenticationQR.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { Component } from 'react'
+import { BallTriangle } from 'react-loader-spinner'
+import {
+  AuthorizationResponseStateStatus,
+  AuthStatusResponse,
+  GenerateAuthRequestURIResponse
+} from './auth-model'
+import {
+  CreateElementArgs,
+  QRType,
+  URIData,
+  ValueResult
+} from '@sphereon/ssi-sdk-qr-react'
+
+import { AuthorizationResponsePayload } from '@sphereon/did-auth-siop'
+import agent from '@components/AuthenticationModal/agent'
+
+export type AuthenticationQRProps = {
+  onAuthRequestRetrieved: () => void
+  onSignInComplete: (payload: AuthorizationResponsePayload) => void
+  setQrCodeData: (text: string) => void
+}
+
+export interface AuthenticationQRState {
+  authRequestURIResponse?: GenerateAuthRequestURIResponse
+  qrCode?: JSX.Element
+}
+
+export default class AuthenticationQR extends Component<AuthenticationQRProps> {
+  state: AuthenticationQRState = {}
+
+  private registerStateSent = false
+  private refreshTimerHandle?: NodeJS.Timeout
+  private qrExpirationMs = 0
+  private timedOutRequestMappings: Set<AuthenticationQRState> =
+    new Set<AuthenticationQRState>()
+
+  private _isMounted = false
+
+  private readonly definitionId = process.env.NEXT_PUBLIC_PRESENTATION_DEF_ID
+
+  componentDidMount() {
+    this.qrExpirationMs =
+      parseInt(process.env.NEXT_PUBLIC_QR_CODE_EXPIRES_AFTER_SEC ?? '120') *
```

Comment:
Make it clear in the ENV vars that we are talking about SSI, to not have it confused with other ENV vars

Reply:
Fixed